soccer/gameplay/plays/offense/distraction1.py

```diff
@@ -0,0 +1,188 @@
+                        behavior.Behavior.State.running)
+        self.add_state(distraction1.State.centerpass,
+                        behavior.Behavior.State.running)
+        self.add_state(distraction1.State.passing,
+                        behavior.Behavior.State.running)
+        self.add_state(distraction1.State.cross,
+                        behavior.Behavior.State.running)
+        self.add_state(distraction1.State.shoot,
+                        behavior.Behavior.State.running)
+        
+        
+        self.add_transition(behavior.Behavior.State.start,
+                        distraction1.State.setup, lambda: True, 'immediately')
+
+        self.add_transition(distraction1.State.setup,
+                        distraction1.State.passing, lambda: not self.ball_is_far,'1-passing')
+
+        #optionaladjustment can be skipped if ball is not far from goal
+        self.add_transition(distraction1.State.setup, 
+                        distraction1.State.optionaladjustment, lambda: self.ball_is_far and 
+                        self.subbehavior_with_name('striker moves').is_done_running(),'1-2')
+
+        self.add_transition(distraction1.State.optionaladjustment,
+                        distraction1.State.centerpass, lambda: self.subbehavior_with_name('move half').is_done_running() and 
+                        self.subbehavior_with_name('move to distract').is_done_running() , '2-centerpass')
+
+        self.add_transition(distraction1.State.centerpass,
+                        distraction1.State.passing, lambda: self.subbehavior_with_name('center pass').is_done_running(), 'centerpass-passing')
+
+        self.add_transition(distraction1.State.passing, 
+                        distraction1.State.cross, lambda: (self.has_subbehavior_with_name('distract pass') and self.subbehavior_with_name('distract pass').is_done_running() )
+                        , 'passing-crossing')
+
+        self.add_transition(distraction1.State.passing,
+                        distraction1.State.shoot, lambda: (self.has_subbehavior_with_name('striker pass') and self.subbehavior_with_name('striker pass').is_done_running() ),'pass-shoot' )
+
+        self.add_transition(distraction1.State.cross,
+                        distraction1.State.shoot, lambda: self.subbehavior_with_name('pass to striker').is_done_running()
+                        , 'crossing-shooting')
+
+        self.add_transition(distraction1.State.shoot,
+                        distraction1.State.setup, lambda: self.has_subbehavior_with_name('shooting') and self.subbehavior_with_name('shooting').is_done_running(), 'repeat')
+
+        
+        self.d1 = robocup.Point(0.40*constants.Field.Width, 0.95*constants.Field.Length) #the first distraction point
+        self.d2 = robocup.Point(0.40*constants.Field.Width, 0.8*constants.Field.Length) #the second distraction point
+        self.s1 = robocup.Point(-0.40*constants.Field.Width, 0.9*constants.Field.Length) #striker's position
```

Comment:
If these are never changing you can add them as constants in the top. There's another comment somewhere with the link to the example